newIDE/app/src/EventsSheet/EventsTree/InstructionsList.js

```diff
@@ -69,6 +69,19 @@ export default class InstructionsList extends React.Component<Props, State> {
       });
   };
 
+  onQuickAddNewInstruction = (e: any) => {
+    if (e.button !== 2) return;
```

Comment:
We won't need this with `onContextMenu` - in general prefer to avoid hardcoding mouse button identifiers because on touch enabled device, context menu can be enabled by other things than a right click (a long press can do the same job on a tablet) and check callbacks offered by the UI components (i.e: Material UI components) or React itself if you're dealing with a div/span/button/etc.

Reply:
Will have another look. I was wondering if we would want to have this available on touch devices which are likely to be small, but if we have some mechanic to detect long press I could add it?

Reply:
Doesn't onContextMenu work for the right click?

Reply:
I will give it a try tomorrow 😊 thank you

Reply:
@4ian onContextMenu is already used to trigger back at the event sheet instructionsListContextMenu. Is it ok to completely overwrite that? Not sure exactly what that does, apart of making a menu with undo/redo commands in it

Reply:
ok I refactored it to use the already established mechanism with onContextMenu